File: src/main/java/minic/ui/MiniCLexerOverlaySegment.java
package minic.ui;

import java.util.Objects;

/**
 * Lexer 遮罩视图中的一段源码文本。
 *
 * @param text 源码文本片段
 * @param active 是否覆盖当前 token
 */
public record MiniCLexerOverlaySegment(String text, boolean active) {
    public MiniCLexerOverlaySegment {
        Objects.requireNonNull(text, "text");
    }
}

File: uiweb/src/visual/MiniCLexerOverlaySegment.ts
import type { JavaMirrorFile } from "../translation/javaMirror";

export const miniCLexerOverlaySegmentMirror = {
  javaPath: "src/main/java/minic/uilocal/visual/MiniCLexerOverlaySegment.java",
  webPath: "uiweb/src/visual/MiniCLexerOverlaySegment.ts",
  packageName: "minic.uilocal.visual",
  exportName: "MiniCLexerOverlaySegment",
  kind: "record",
  imports: [
    "java.util.Objects"
  ],
  fields: [],
  methods: [],
} as const satisfies JavaMirrorFile;

export class MiniCLexerOverlaySegment {
  static readonly mirror = miniCLexerOverlaySegmentMirror;

  constructor(readonly text: string, readonly active: boolean) {}
}

export default MiniCLexerOverlaySegment;
